File: 백준/Silver/1966. 프린터 큐/프린터 큐.java
import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int t = Integer.parseInt(br.readLine());

        for (int i = 0; i < t; i++) {
            st = new StringTokenizer(br.readLine());
            int n = Integer.parseInt(st.nextToken());
            int m = Integer.parseInt(st.nextToken());
            st = new StringTokenizer(br.readLine());
            int[] works = new int[n];
            int[] importance = new int[n];
       
            for (int j = 0; j < n; j++) {
                int work = Integer.parseInt(st.nextToken());
                works[j] = work;
                importance[j] = work;
            }
            Arrays.sort(importance);

            int cur = 0;
            int cnt = 1;
            int idx = n-1;
            while (true) {
                int num = works[cur % n];
                if (num == -1) {
                    cur++;
                    continue;
                }
                if (num == importance[idx]) {
                    if (cur % n == m) {
                        System.out.println(cnt);
                        break;
                    }
                    works[cur % n] = -1;
                    idx--;
                    cnt++;
                }
                cur++;
            }
        }
    }
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
# 프린터 큐
import sys
input = sys.stdin.readline
from collections import deque

t = int(input())

for i in range(t):
    n, m = map(int,input().split())
    w = list(map(int,input().split()))
    works = []
    importance = []

    for j in range(n):
        work = w[j]
        works.append((j, work))
        importance.append(work)

    importance.sort()
    que = deque(works)
    cnt = 1
    
    while que:
        idx, num = que.popleft()
        if num < importance[-1]:
            que.append((idx, num))
        else:
            importance.pop()
            if idx == m:
                print(cnt)
                break
            cnt += 1
